banzai_floyds/tests/test_extract.py

```diff
@@ -1,108 +1,66 @@
-                           fake_frame.orders, wavelength_bins)
-    fitted_widths = fit_profile_width(binned_data, fake_frame.input_profile_centers)
-    for fitted_width, bins in zip(fitted_widths, wavelength_bins):
-        x = np.arange(bins['center'][0], bins['center'][-1] + 1)
-        np.testing.assert_allclose(fitted_width(x), fwhm_to_sigma(fake_frame.input_profile_width), rtol=0.03)
-
-
-def test_background_fitting():
-    np.random.seed(234515)
-    fake_frame = generate_fake_science_frame(include_sky=True)
-    wavelength_bins = get_wavelength_bins(fake_frame.wavelengths)
-    binned_data = bin_data(fake_frame.data, fake_frame.uncertainty, fake_frame.wavelengths,
-                           fake_frame.orders, wavelength_bins)
-    fake_profile_width_funcs = [lambda _: fwhm_to_sigma(fake_frame.input_profile_width)
-                                for _ in fake_frame.input_profile_centers]
-    fitted_background = fit_background(binned_data, fake_frame.input_profile_centers, fake_profile_width_funcs)
-    fake_frame.background = fitted_background
-    binned_fitted_background = bin_data(fake_frame.background, fake_frame.uncertainty, fake_frame.wavelengths,
-                                        fake_frame.orders, wavelength_bins)
-    binned_input_sky = bin_data(fake_frame.input_sky, fake_frame.uncertainty, fake_frame.wavelengths,
-                                fake_frame.orders, wavelength_bins)
-    np.testing.assert_allclose(binned_fitted_background['data'].groups.aggregate(np.sum),
-                               binned_input_sky['data'].groups.aggregate(np.sum),
-                               rtol=0.033)
+def test_extraction_region():
+    FakeImage = namedtuple('FakeImage', ['binned_data', 'meta', 'extraction_windows'])
+    nx, ny = 103, 101
+    x, y = np.meshgrid(np.arange(nx), np.arange(ny))
+    order_centers = [20, 60]
+    order_height = 21
+    orders = np.zeros_like(x)
+    for order_id in [1, 2]:
+        in_order = order_centers[order_id - 1] - order_height // 2 <= y
+        in_order = np.logical_and(y <= order_centers[order_id - 1] + order_height // 2, in_order)
+        orders[in_order] = order_id
+    profile_sigma = 1.0
+    y_profile = np.zeros_like(x)
+    for order_id in [1, 2]:
+        y_profile[orders == order_id] = y[orders == order_id] - order_centers[order_id - 1]
+    binned_data = Table({'x': x.ravel(), 'y': y.ravel(), 'order': orders.ravel(),
+                         'profile_sigma': profile_sigma * np.ones(x.size),
+                         'y_profile': y_profile.ravel()})
+    fake_data = FakeImage(binned_data, {}, [[-5.0, 5.0], [-5.0, 5.0]])
+    set_extraction_region(fake_data)
+    # The extraction should be +- 5 pixels high so there should be 11 pixels in the extraction region
+    for order in [1, 2]:
+        in_order = fake_data.binned_data['order'] == order
+        assert np.sum(fake_data.binned_data['extraction_window'][in_order]) == 11 * nx
 
 
 def test_extraction():
```

Comment:
We've removed a lot of the dependencies here when testing for extraction. This should be equivalent albeit more clean method to extraction.